tests/unit/swagger/templateHelpers.spec.ts

```diff
@@ -264,6 +264,76 @@ describe('ValidationService', () => {
     });
   });
 
+  describe('Enum validate', () => {
+    type Enumeration = string[] | number[];
+
+    it('should enum number value', () => {
+      const name = 'name';
+      const value = '1';
+      const error = {};
+      const enumeration: Enumeration = ['0', '1'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(value);
+    });
+
+    it('should enum string value', () => {
+      const name = 'name';
+      const value = 'HELLO';
+      const error = {};
+      const enumeration: Enumeration = ['HELLO'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(value);
+    });
+
+    it('should enum no member', () => {
+      const name = 'name';
+      const value = 'HI';
+      const error: any = {};
+      const enumeration: Enumeration = [];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(undefined);
+      expect(error[name].message).to.equal(`no member`);
+    });
+
+    it('should enum out of member', () => {
+      const name = 'name';
+      const value = 'SAY';
+      const error: any = {};
+      const enumeration: Enumeration = ['HELLO', 'HI'];
+      const result = new ValidationService({}).validateEnum(name, value, error, enumeration);
+      expect(result).to.equal(undefined);
+      expect(error[name].message).to.equal(`should be one of the following; ['HELLO', 'HI']`);
+    });
+
+    it('accepts a string value of a numeric enum', () => {
```

Comment:
Can you please make this a negative test case?

In other words, can you make it so that submitting `1` is a failure if the enum contains `”1”`?

I’m asking for this because `”1”.toLowerCase()` is a function but `1.toLowerCase()` is not. So if we let an string replace a number (and vica-versa) then we put tsoa users at risk of a runtime error.